### PR TITLE
Storage

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

reopens #1 after renaming default branch and updating `main` in #3 

## Why?

its useful!

### CODE REVIEW

File: src/storage.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+		return this.available;
+	},
+
+	get(key: string): unknown | null {
+		if (!this.available) {
+			return;
+		}
+
+		let data;
```
Comment (SiAdcock) [inline src/storage.ts:29]: Is it worth trying to assign a type to this

```ts
type Data = {value: unknown; expires: string | number | Date | undefined} | null | undefined

let data: Data
```

A bit hairy?

File: src/storage.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
+const store = (type: 'sessionStorage' | 'localStorage') => ({
+	available: undefined as boolean | undefined,
+	storage: window[type],
+	isAvailable(): boolean | undefined {
```
Comment (SiAdcock) [inline src/storage.ts:4]: When does / should this be called?

**Comment by SiAdcock** [comment]
Looks great! Do you think it's worth putting together some quick docs for it?

**Comment by sndrs** [comment]
closing in favour of #31

**Pull request closed without merging**